### ISSUE TITLE
generate example cases when using hypothesis

### ISSUE THREAD

**Issue by charles-cooper** [vyperlang/titanoboa#31]
we should have a special strategy which uses strategies taken from the state or contract storage. e.g. in the following example,
```vyper
def __init__():
    self.foo = msg.sender
```
```python
@given(addr=st.addresses)  # (?)
def test_bar(addr):
    ...
```
st.addresses should include whatever `self.foo` is as an example.

**Comment by bout3fiddy** [comment]
So if I am to understand this, the strategy would change the storage var `foo`

**Comment by charles-cooper** [comment]
> So if I am to understand this, the strategy would change the storage var `foo`

not exactly. it would insert hypothesis examples, so this one would be similar to
```
@given(...)
@example(contract._storage.foo.get())
def test_bar(addr):
    ...
```

**Comment by charles-cooper** [comment]
i don't think #222 implements this exactly, although it's similar
